### ISSUE TITLE
dump with arangodump v3.x cant be restored with arangorestore v.2.8.x

### ISSUE THREAD

**Issue by baslr** [arangodb/arangodb#1848]
```
./arangorestore (v2.8.x) --collection tags --server.endpoint tcp://:13370 --input-directory ../dump/
Please specify a password:
Server version: 2.8.0
# Connected to ArangoDB 'tcp://:13370'
# Re-creating document collection 'tags'...
# Loading data into document collection 'tags'...
Error: bad parameter: got error from server: HTTP 500 (Internal Server Error): ArangoError 400: bad parameter: received invalid JSON data for collection 16740495
Processed 1 collection(s), read 6639 byte(s) from datafiles, sent 1 batch(es)

# Example data
# 2.8 dump
{"type":2300,"key":"22769807","rev":"22769807","data":{"_key":"22769807","_rev":"22769807","t":"vTKXhUJbWB"}}

# 3.x dump
{"type":2300,"data":{"_id":"tags/Animation","_key":"Animation","_rev":"1390899"}}

# modified dump: added key and rev
{"type":2300,"key":"Animation","rev":"1390899","data":{"_id":"tags/Animation","_key":"Animation","_rev":"1390899"}}

./arangorestore (v.2.8.x) --collection tags --server.endpoint tcp://:13370 --input-directory ../dump/
Please specify a password:
Server version: 2.8.0
# Connected to ArangoDB 'tcp://:13370'
# Re-creating document collection 'tags'...
# Loading data into document collection 'tags'...
Processed 1 collection(s), read 116 byte(s) from datafiles, sent 1 batch(es)

also arangorestore v3 wont restore into a 2.8 server

./arangorestore (v3.x) --server.endpoint tcp://:13370 --collection tags --input-directory dump
Please specify a password:
Server version: 2.8.0
2016-05-31T23:22:34Z [93194] ERROR got incompatible server version '2.8.0'
2016-05-31T23:22:34Z [93194] FATAL giving up!

# arangorestore restores dump created with arangodump v2
/arangorestore (v3.x) --server.endpoint tcp://:8001 --collection test --input-directory dumptest
Server version: 3.0.x-devel
# Connected to ArangoDB 'http+tcp://:8001'
# Re-creating document collection 'test'...
# Loading data into document collection 'test'...
Processed 1 collection(s), read 1210 byte(s) from datafiles, sent 1 batch(es)
```
Conclusion:
`arangodump v3` needs some compatibility flag to dump with `key` and `rev`.

```
./arangodump (v3) --help
Section 'global options' (Global configuration)
  --dump-format <uint64>                    dump version format '2' xor '3' (default: 3)

./arangodump (v3) --dump-format 2 --collection test --server.endpoint tcp://:13370 --output-directory dumptest
```
\ | restore v2 | restore v3
------------ | ----| ---------
dump v2 | x | x
dump v3 | - | x

**Comment by baslr** [comment]
In addition: if its a edge collection v2.8 requires `type:2301` instead of `type:2300`
```
# restore with type:2300
./arangorestore (v2.8) --server.endpoint tcp://:13370 --input-directory dumptest --collection test --server.disable-authentication true
Server version: 2.8.1
# Connected to ArangoDB 'tcp://:13370'
# Re-creating edge collection 'test'...
# Loading data into edge collection 'test'...
Error: collection type invalid: got error from server: HTTP 400 (Bad Request): ArangoError 1218: collection type invalid: collection type invalid: expecting document collection, got edge collection
Processed 1 collection(s), read 242 byte(s) from datafiles, sent 1 batch(es)

# restore with type:2301
./arangorestore --server.endpoint tcp://:13370 --input-directory dumptest --collection test --server.disable-authentication true
Server version: 2.8.1
# Connected to ArangoDB 'tcp://:13370'
# Re-creating edge collection 'test'...
# Loading data into edge collection 'test'...
Processed 1 collection(s), read 242 byte(s) from datafiles, sent 1 batch(es)
```

**Comment by jsteemann** [comment]
I can confirm that restoring a 3.0 dump into a 2.8 does not work because of the changed format.

**Comment by jsteemann** [comment]
arangodump in 3.0 now has an option `--compat28` that can be set to `true` when invoking arangodump.
This will produce a JSON format that is compatible with 2.8.

**Comment by jsteemann** [comment]
Should be included in the 3.0rc2
